docs/plus/rbac.md

```diff
@@ -0,0 +1,120 @@
+# Role Based Access Control
+
+Role Based Access Control(RBAC) helps you restrict actions that can be performed on specific Kubernetes resources. To make this possible, RBAC lets you create roles with rules which define access permissions for your specified resource.
+
+These roles can then be binded to Kubernetes subjects, which could be User, Group or ServiceAccount. 
+
+!!! note
+    Rules or permissions are purely additive and there are no deny rules.
+
+Now, there are two types of roles available,
+* Role: These set permissions within a particular namespace i.e. is for namespaced resources, like, pods, deployments.
+* ClusterRole: These set permissions for non-namespaced resources, like, nodes, urls.
+
+and, similarly there are two types of binding available,
+* RoleBinding: These grant permissions within a specific namespace.
+* ClusterRoleBinding: These grant cluster wide permissions .
+
+!!! tip "Learn more"
+    * [Role API Reference](../reference/cdk8s-plus-24/typescript.md#role)
+    * [RoleBinding API Reference](../reference/cdk8s-plus-24/typescript.md#role-binding)
+
+## Role
+
+### Create role and add rules to it
+
+```typescript
+import * as kplus from 'cdk8s-plus-24';
+import { Construct } from 'constructs';
+import { App, Chart, ChartProps } from 'cdk8s';
+
+export class MyChart extends Chart {
+  constructor(scope: Construct, id: string, props: ChartProps = { }) {
+    super(scope, id, props);
+
+    // Creating RBAC Role
+    const role = new kplus.Role(this, 'SampleRole');
+
+    // Adding list of rules to the Role for 'deployments' namespaced resource
+    role.allow(
+      ['get', 'list', 'watch', 'create', 'update', 'patch', 'delete'],
+      kplus.ApiResource.custom({
+        apiGroup: 'apps',
+        resourceType: 'deployments',
+      })
+    );
+  }
+}
+
+const app = new App();
+new MyChart(app, 'rbac-docs');
+app.synth();
+```
+
+### RoleBinding with User
+
+```typescript
+const user = kplus.User.fromName(this, 'SampleUser', 'Jane');
+const binding = role.bind(user);
+```
+
+### RoleBinding with Group
+
+```typescript
+const group = kplus.Group.fromName(this, 'SampleGroup', 'sample-group');
+const binding = role.bind(group);
+```
+
+### RoleBinding with ServiceAccount
```

Comment:
All of these should be inlined within the previous code example where you create the role. Just put some code comments above each use-case...

```ts
// you can bind this role to a specific group, user, or service account
// this will grant these permissions to the appropriate entity (subject)
role.bind(kplus.User.fromName(this, 'SampleUser', 'Jane'));
role.bind(kplus.Group.fromName(this, 'SampleGroup', 'sample-group'));
role.bind(new kplus.ServiceAccount(this, 'SampleServiceAccount'));
```